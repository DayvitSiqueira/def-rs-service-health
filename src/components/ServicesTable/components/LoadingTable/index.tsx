function LoadingTable() {
  return (
    <table className="w-full text-sm text-left text-gray-500 dark:text-gray-400">
      <thead className="text-xs text-gray-700 uppercase dark:text-gray-400 text-center">
        <tr>
          {[0, 1, 2].map(item => {
            return (
              <th
                key={item}
                scope="col"
                className="p-4 whitespace-nowrap bg-gray-100 dark:bg-gray-800 border-b-2 h-12"
              />
            )
          })}
        </tr>
      </thead>

      <tbody>
        <tr>
          {[0, 1, 2].map(item => (
            <td key={item} className="p-4">
              <div role="status" className="animate-pulse">
                <div className="h-11 bg-gray-200 rounded-lg dark:bg-gray-700 w-90 mb-5" />
                <div className="h-11 bg-gray-200 rounded-lg dark:bg-gray-700 w-90 mb-5" />
                <div className="h-11 bg-gray-200 rounded-lg dark:bg-gray-700 w-90 mb-2" />
              </div>
            </td>
          ))}
        </tr>
      </tbody>
    </table>
  )
}

export default LoadingTable
