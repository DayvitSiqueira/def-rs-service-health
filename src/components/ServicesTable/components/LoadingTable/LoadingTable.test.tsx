import React from 'react'
import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import LoadingTable from './index'

describe('LoadingTable', () => {
  it('renders the LoadingTable component correctly', () => {
    render(<LoadingTable />)
    const tableElement = screen.getByRole('table')
    const thElements = screen.getAllByRole('columnheader')
    const tdElements = screen.getAllByRole('cell')
    const statusElements = screen.getAllByRole('status')

    expect(tableElement).toBeInTheDocument()
    expect(thElements.length).toBe(3)
    expect(tdElements.length).toBe(3)
    expect(statusElements.length).toBe(3)
  })

  it('matches the snapshot', () => {
    const { container } = render(<LoadingTable />)
    expect(container).toMatchSnapshot()
  })
})
