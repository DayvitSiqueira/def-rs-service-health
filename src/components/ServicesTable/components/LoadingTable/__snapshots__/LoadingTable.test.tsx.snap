// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingTable Component matches the snapshot 1`] = `
<div>
  <table
    class="w-full text-sm text-left text-gray-500 dark:text-gray-400"
  >
    <thead
      class="text-xs text-gray-700 uppercase dark:text-gray-400 text-center"
    >
      <tr>
        <th
          class="p-4 whitespace-nowrap bg-gray-100 dark:bg-gray-800 border-b-2 h-12"
          scope="col"
        />
        <th
          class="p-4 whitespace-nowrap bg-gray-100 dark:bg-gray-800 border-b-2 h-12"
          scope="col"
        />
        <th
          class="p-4 whitespace-nowrap bg-gray-100 dark:bg-gray-800 border-b-2 h-12"
          scope="col"
        />
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          class="p-4"
        >
          <div
            class="animate-pulse"
            role="status"
          >
            <div
              class="h-11 bg-gray-200 rounded-lg dark:bg-gray-700 w-90 mb-5"
            />
            <div
              class="h-11 bg-gray-200 rounded-lg dark:bg-gray-700 w-90 mb-5"
            />
            <div
              class="h-11 bg-gray-200 rounded-lg dark:bg-gray-700 w-90 mb-2"
            />
          </div>
        </td>
        <td
          class="p-4"
        >
          <div
            class="animate-pulse"
            role="status"
          >
            <div
              class="h-11 bg-gray-200 rounded-lg dark:bg-gray-700 w-90 mb-5"
            />
            <div
              class="h-11 bg-gray-200 rounded-lg dark:bg-gray-700 w-90 mb-5"
            />
            <div
              class="h-11 bg-gray-200 rounded-lg dark:bg-gray-700 w-90 mb-2"
            />
          </div>
        </td>
        <td
          class="p-4"
        >
          <div
            class="animate-pulse"
            role="status"
          >
            <div
              class="h-11 bg-gray-200 rounded-lg dark:bg-gray-700 w-90 mb-5"
            />
            <div
              class="h-11 bg-gray-200 rounded-lg dark:bg-gray-700 w-90 mb-5"
            />
            <div
              class="h-11 bg-gray-200 rounded-lg dark:bg-gray-700 w-90 mb-2"
            />
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`LoadingTable matches the snapshot 1`] = `
<div>
  <table
    class="w-full text-sm text-left text-gray-500 dark:text-gray-400"
  >
    <thead
      class="text-xs text-gray-700 uppercase dark:text-gray-400 text-center"
    >
      <tr>
        <th
          class="p-4 whitespace-nowrap bg-gray-100 dark:bg-gray-800 border-b-2 h-12"
          scope="col"
        />
        <th
          class="p-4 whitespace-nowrap bg-gray-100 dark:bg-gray-800 border-b-2 h-12"
          scope="col"
        />
        <th
          class="p-4 whitespace-nowrap bg-gray-100 dark:bg-gray-800 border-b-2 h-12"
          scope="col"
        />
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          class="p-4"
        >
          <div
            class="animate-pulse"
            role="status"
          >
            <div
              class="h-11 bg-gray-200 rounded-lg dark:bg-gray-700 w-90 mb-5"
            />
            <div
              class="h-11 bg-gray-200 rounded-lg dark:bg-gray-700 w-90 mb-5"
            />
            <div
              class="h-11 bg-gray-200 rounded-lg dark:bg-gray-700 w-90 mb-2"
            />
          </div>
        </td>
        <td
          class="p-4"
        >
          <div
            class="animate-pulse"
            role="status"
          >
            <div
              class="h-11 bg-gray-200 rounded-lg dark:bg-gray-700 w-90 mb-5"
            />
            <div
              class="h-11 bg-gray-200 rounded-lg dark:bg-gray-700 w-90 mb-5"
            />
            <div
              class="h-11 bg-gray-200 rounded-lg dark:bg-gray-700 w-90 mb-2"
            />
          </div>
        </td>
        <td
          class="p-4"
        >
          <div
            class="animate-pulse"
            role="status"
          >
            <div
              class="h-11 bg-gray-200 rounded-lg dark:bg-gray-700 w-90 mb-5"
            />
            <div
              class="h-11 bg-gray-200 rounded-lg dark:bg-gray-700 w-90 mb-5"
            />
            <div
              class="h-11 bg-gray-200 rounded-lg dark:bg-gray-700 w-90 mb-2"
            />
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
